internal/analytics/svc/analytics.go

```diff
@@ -0,0 +1,204 @@
+package svc
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime/debug"
+	"sync"
+
+	anaConsts "github.com/ActiveState/cli/internal/analytics/constants"
+	"github.com/ActiveState/cli/internal/condition"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/machineid"
+	"github.com/ActiveState/cli/internal/singleton/uniqid"
+	"github.com/ActiveState/cli/internal/updater"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	ga "github.com/ActiveState/go-ogle-analytics"
+	"github.com/ActiveState/sysinfo"
+)
+
+// Analytics instances send analytics events to GA and S3 endpoints without delay. It is only supposed to be used inside the `state-svc`.  All other processes should use the DefaultClient.
+type Analytics struct {
+	gaClient         *ga.Client
+	customDimensions *CustomDimensions
+	eventWaitGroup   *sync.WaitGroup
+}
+
+// New initializes the analytics instance with all custom dimensions known at this time
+func New() *Analytics {
+	r := &Analytics{
+		eventWaitGroup: &sync.WaitGroup{},
+	}
+
+	return r
+}
+
+func (a *Analytics) Configure(cfg *config.Instance, auth *authentication.Auth) {
+	installSource, err := storage.InstallSource()
+	if err != nil {
+		logging.Error("Could not detect installSource: %s", errs.Join(err, " :: ").Error())
+	}
+
+	id := machineid.UniqID()
+	if id == "unknown" {
```

Comment:
Where do we return "unknown"? Seems the value used for this case is "11111111-1111-1111-1111-111111111111".

Reply:
Ha.  I just copied that from the original file.  But that was outdated, I guess...  Will fix.